examples/seismic/model.py

```diff
@@ -263,27 +260,40 @@ def demo_model(preset, **kwargs):
 def damp_boundary(damp, nbpml, spacing):
     """Initialise damping field with an absorbing PML layer.
 
-    :param damp: Array data defining the damping field
-    :param nbpml: Number of points in the damping layer
-    :param spacing: Grid spacing coefficent
+    :param damp: The :class:`Function` for the damping field.
+    :param nbpml: Number of points in the damping layer.
+    :param spacing: Grid spacing coefficent.
     """
     dampcoeff = 1.5 * np.log(1.0 / 0.001) / (40.)
-    ndim = len(damp.shape)
-    for i in range(nbpml):
+    assert all(damp._offset_domain[0] == i for i in damp._offset_domain)
+    for i in range(nbpml + damp._offset_domain.left[0]):
         pos = np.abs((nbpml - i + 1) / float(nbpml))
         val = dampcoeff * (pos - np.sin(2*np.pi*pos)/(2*np.pi))
-        if ndim == 2:
-            damp[i, :] += val/spacing[0]
-            damp[-(i + 1), :] += val/spacing[0]
-            damp[:, i] += val/spacing[1]
-            damp[:, -(i + 1)] += val/spacing[1]
+        if damp.ndim == 2:
+            damp.data_with_halo[i, :] += val/spacing[0]
+            damp.data_with_halo[-(i + 1), :] += val/spacing[0]
+            damp.data_with_halo[:, i] += val/spacing[1]
+            damp.data_with_halo[:, -(i + 1)] += val/spacing[1]
```

Comment:
Can't we assign directly to `data` that has already the offset?

Reply:
maybe? I'm open to contributions here... I just tried to emulate what's in ``master``.

Reply:
Tried it, seems to work the right way with `dam.data` and `range(nbpml)`